go-controller/cmd/ovnkube-identity/ovnkubeidentity.go

```diff
@@ -0,0 +1,369 @@
+	}
+	webhookMux.Handle("/node", nodeHandler)
+
+	// in non-ic ovnkube-node does not have the permissions to update pods
+	if cliCfg.enableInterconnect {
+		informerFactory := informers.NewSharedInformerFactory(client, 10*time.Minute)
+		nodeInformer := informerFactory.Core().V1().Nodes().Informer()
+		informerFactory.Start(stopCh)
+		klog.Infof("Waiting for caches to sync")
+		cache.WaitForCacheSync(c.Context.Done(), nodeInformer.HasSynced)
+
+		nodeLister := listers.NewNodeLister(nodeInformer.GetIndexer())
+		podWebhook := admission.WithCustomValidator(
+			scheme.Scheme,
+			&corev1.Pod{},
+			ovnwebhook.NewPodAdmissionWebhook(nodeLister),
+		).WithRecoverPanic(true)
+		podHandler, err := admission.StandaloneWebhook(
+			podWebhook,
+			admission.StandaloneOptions{
+				Logger:      logger.WithName("pod.network-identity"),
+				MetricsPath: "pod.network-identity",
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to setup the pod admission webhook: %w", err)
+		}
+		webhookMux.Handle("/pod", podHandler)
+	}
+
+	cfg := &tls.Config{
+		NextProtos: []string{"h2"},
+		MinVersion: tls.VersionTLS10,
+	}
+
+	certPath := filepath.Join(cliCfg.certDir, "tls.crt")
+	keyPath := filepath.Join(cliCfg.certDir, "tls.key")
+	certWatcher, err := certwatcher.New(certPath, keyPath)
+	if err != nil {
+		return fmt.Errorf("failed to setup certwatcher: %v", err)
+	}
+	cfg.GetCertificate = certWatcher.GetCertificate
+
+	go func() {
+		if err := certWatcher.Start(c.Context); err != nil {
+			klog.Errorf("Certificate watcher failed to start: %v", err)
+		}
+	}()
+	srv := &http.Server{
+		Handler:           webhookMux,
+		IdleTimeout:       90 * time.Second,
+		ReadHeaderTimeout: 32 * time.Second,
+	}
+
+	l := net.ListenConfig{
+		Control: func(network, address string, c syscall.RawConn) error {
+			return c.Control(func(fd uintptr) {
+				// Enable SO_REUSEPORT
+				err := syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, unix.SO_REUSEPORT, 1)
+				if err != nil {
+					fmt.Println("Failed to set SO_REUSEPORT:", err)
+				}
+			})
+		},
+	}
+
+	idleWebhookConnectionsClosed := make(chan struct{})
+	defer func() {
+		klog.Infof("Waiting for the webhook server to gracefully close")
+		<-idleWebhookConnectionsClosed
+	}()
+	innerListener, err := l.Listen(c.Context, "tcp", net.JoinHostPort(cliCfg.host, strconv.Itoa(cliCfg.port)))
+	if err != nil {
+		return fmt.Errorf("failed to create the listener: %v", err)
+	}
+	listener := tls.NewListener(innerListener, cfg)
+
+	klog.Infof("Starting the webhook server")
+	go func() {
+		<-c.Context.Done()
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+		defer cancel()
+		if err := srv.Shutdown(ctx); err != nil {
+			klog.Errorf("Failed shutting down the HTTP server: %v", err)
+		}
+		close(idleWebhookConnectionsClosed)
+	}()
+
+	return srv.Serve(listener)
+}
+
+func runClusterManager(ctx context.Context, leaderID string, restCfg *rest.Config) error {
```

Comment:
i wonder if calling this cluster manager is going to be confusing with our other cluster manager. Maybe we should call it admissionCertManager or something?